component/status/notifications.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status // import "go.opentelemetry.io/collector/component/status"
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/multierr"
+)
+
+var errRegistrationNotFound = errors.New("registration not found")
+
+// Notifications is a mechanism that facilitates the passing of status events between components.
+// Status events can be used to indicate that a component is functioning as expected, or that it is
+// in an error state. Components can send status events using the ReportStatus function, and
+// components interested in receiving events can register a listener via the RegisterListener
+// function.
+type Notifications struct {
```

Comment:
why does this need to be public? Can it be in an internal package for the moment?

Reply:
It's public so it can be used in the service package. I think I could move it to the service if I made status.Listener public. Would that move make sense?

Reply:
you can also move to a directory that includes "internal" in the path (see https://go.dev/doc/go1.4#internalpackages)

Reply:
It is now internal within the service subtree.